src/uint/bits.rs

```diff
@@ -1,6 +1,13 @@
 use crate::{Limb, UInt, Word};
 
 impl<const LIMBS: usize> UInt<LIMBS> {
+    /// Get the value of the bit at position `index`, as a 0- or 1-valued Word.
+    #[inline]
+    pub const fn bit(self, index: usize) -> Word {
```

Comment:
It might be good to name this `bit_vartime` to note that the access is variable time with respect to `index`, since it effectively computes a pointer based on its value.

Reply:
Variable-index table lookups are a classical source of microarchitectural sidechannel attacks, e.g. cache timing attacks

Reply:
I should probably also note a fully constant-time version is also possible but it would be significantly slower: you'd need to scan over every word every time, accumulating the result as a `Word`, and masking the value with bitwise arithmetic if the index is not within a given word

Reply:
Yes I considered that, but I can't think of a use case right now. I suppose `bit_vartime` it is. In my case I'm just iterating through all the bits so there's no security concern.

I should note that the `bits` method below does not appear to be constant time either due to the initial iteration over the zero-valued limbs.

Reply:
Yeah good observation re: `bits` cc @mikelodder7

Reply:
It’s not supposed to be constant time. Renaming is fine.

Reply:
Is there a reason to make this constant time? To make this constant time requires touching every bit which doesn’t seem necessary

Reply:
@mikelodder7 nah, I'm just more concerned with naming and making it clear when functions have non-constant-time behavior

Reply:
Got it. Fair enough